File: Java8037/src/com/ssuhanov/GameStart.java
package com.ssuhanov;

/**
 * Created by ssuhanov on 4/15/17.
 */
public class GameStart {

    private static final String GoodByeMessage = "Good bye.\n";
    
    private String exitWord;
    private Game game;

    public GameStart(String exitWord) {
        this.exitWord = exitWord;
        this.game = Game8037.Create();
    }

    void startGame() {
        if (game != null) {
            String inputWord;

            do {
                game.greet(exitWord);
                inputWord = readLine();
            } while (checkInputWord(inputWord));
        }
    }

    private boolean checkInputWord(String inputWord) {
        boolean isExitWord = inputWord.equals(exitWord);
        if (isExitWord) {
            System.out.println(GoodByeMessage);
        } else {
            game.checkUserAnswer(inputWord);
        }
        return !isExitWord;
    }

    private String readLine() {
        java.util.Scanner scanner = new java.util.Scanner(System.in);
        return scanner.next();
    }

}

File: CSharp8037/CSharp8037/GameStart.cs
using System;

namespace CSharp8037
{
	public class GameStart
	{
		const string GreetingMessage = "Please enter your answer (enter \"{0}\" for exit):";
		const string InputAcceptMessage = "You entered \"{0}\".\n";
		const string GoodByeMessage = "Good bye.\n";

		readonly string exitWord;
		readonly IGame iGame;

		public GameStart(string exitWord)
		{
			this.exitWord = exitWord;
			iGame = Game8037.Create();
		}

		public void startGame()
		{
			if (iGame != null)
			{
				string inputWord;
				do
				{
					iGame.greet(exitWord);
					inputWord = Console.ReadLine();
				}
				while (checkInputWord(inputWord));
			}
		}

		bool checkInputWord(string inputWord)
		{
			bool isExitWord = inputWord.Equals(exitWord);
			if (isExitWord)
			{
				Console.WriteLine(GoodByeMessage);
			}
			else
			{
				iGame.checkUserAnswer(inputWord);
			}
			return !isExitWord;
		}
	}
}
